src/runtime_src/core/edge/drm/zocl/zocl_ctrl_ert.c

```diff
@@ -241,20 +246,36 @@ static int zert_create_scu(struct zocl_ctrl_ert *zert, struct xgq_cmd_config_cu
 static void zert_init_cus(struct zocl_ctrl_ert *zert)
 {
 	u32 i = 0;
-	struct zocl_ctrl_ert_cu *cu = &zert->zce_cus[0];
+	struct zocl_ctrl_ert_cu *cu = NULL;
 
+	zert->zce_cus = kzalloc(sizeof(struct zocl_ctrl_ert_cu) * MAX_CUS,
+				GFP_KERNEL);
+	if (!zert->zce_cus) {
+		zert_err(zert, "Memory allocation for CUs are failed");
+		return;
+	}
+	zert->zce_num_cus = MAX_CUS;
```

Comment:
do you still need this member variable - zce_num_cus?

Reply:
Fixed